### DISCUSSION TITLE
Default OLTP protocol in dotnet

### DISCUSSION THREAD

**Discussion by Gladskih** [open-telemetry/opentelemetry-dotnet#5987] [Q&A] [answered]
https://github.com/open-telemetry/opentelemetry-dotnet/blob/core-1.9.0/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md#otlpexporteroptions
Protocol: OTLP transport protocol. The default is OtlpExportProtocol.Grpc.

https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/v1.9.0/docs/config.md#otlp
OTEL_EXPORTER_OTLP_PROTOCOL - Default value: http/protobuf

Is one of the sources wrong? Or what am I missing?

**Comment by cijothomas** [comment, marked as answer]
Auto-instrumentation used http as default, but sdk (i.e this repo) uses gRPC as default.
The spec recommends Http as default *unless* there is a good reason. For this repo, there is *good* reason - it shipped 1.0 with gRPC as default, so changing default to Http is a breaking change. 

This repo should also change to HTTP by default, when there is a major bump (2.* release), but AFAICT, there is no such plans near term.
https://github.com/open-telemetry/opentelemetry-dotnet/issues/2959 is an existing issue for this.

**Reply by Gladskih** [reply]
> sdk (i.e this repo) uses gRPC as default
> This repo should also change to gRPC by default, when ...

Again some confusion. But I guess I got the idea. Thank you.

**Reply by cijothomas** [reply]
Any part that is still confusing? Happy to clarify further.

**Reply by Kielek** [reply]
@cijothomas, please update your previous message `This repo should also change to gRPC` ---> `This repo should also change to HTTP`

**Reply by cijothomas** [reply]
thanks, fixed and added link to the open issue for the same.
